x/evm/keeper/state_transition.go

```diff
@@ -75,39 +75,21 @@ func (k *Keeper) NewEVM(
 
 	// rules are used to determine the active precompiles for the current block height.
 	// i.e rules.IsByzantium, rules.IsConstantinople, etc.
-	rules := cfg.ChainConfig.Rules(big.NewInt(ctx.BlockHeight()), cfg.ChainConfig.MergeNetsplitBlock != nil)
+	rules := cfg.ChainConfig.Rules(big.NewInt(ctx.BlockHeight()), cfg.ChainConfig.MergeNetsplitBlock != nil, 1)
 
-	contracts := make(map[common.Address]vm.PrecompiledContract)
-	active := make([]common.Address, 0)
-
-	// Creates the list of **default** precompiled contracts (not stateful) for this set of rules.
-	// contracts hold the list of all contracts, while active holds the list of all active addresses.
-	// i.e create the list of contracts for Berlin.
-	for addr, c := range vm.DefaultPrecompiles(rules) {
-		contracts[addr] = c
-		active = append(active, addr)
-	}
```

Comment:
nit: why are we removing this part?

Reply:
You no longer need to provide the default precompiles. You only need to provide the stateful precompiles (via `evm.SetStatefulPrecompiles()`